### PR TITLE
Decompile func_ptr_80170008 of w_016

### PR DESCRIPTION
**Pull request by Max-Clark**

### CODE REVIEW

File: include/entity.h [hunk 1]
```diff
@@ -1906,6 +1925,7 @@ typedef union { // offset=0x7C
 SYNC_FIELD(ET_Player, ET_Weapon, anim);
 SYNC_FIELD(ET_Player, ET_WeaponUnk006, anim);
 SYNC_FIELD(ET_Player, ET_WeaponUnk012, anim);
+SYNC_FIELD(ET_Player, ET_WeaponUnk016, anim);
```
Comment (Max-Clark) [inline include/entity.h:1928]: I'm not 100% sure this is necessary, just following the pattern

Comment (Max-Clark) [reply include/entity.h]: It is in fact, not ;)

File: include/entity.h [hunk 2]
```diff
@@ -353,6 +353,24 @@ typedef struct {
...
 } ET_WeaponUnk014;
 
+typedef struct {
+    s16 unk7C;
+    s16 unk7E;
+    s32 unk80;
+    s16 unk84;
+    s32 : 32;
+    s16 : 16;
+    struct Entity* parent;
```
Comment (sozud) [inline include/entity.h:363]: Adding 
```
#ifdef VERSION_PC
u32 padding;
#endif
```
after this point and adding back the assert should fix the issue you were having

Comment (Max-Clark) [reply include/entity.h:363]: Hmmm... I can't get it to work. Don't want to blow all the pipeline time so going to play around on my build machine

File: src/weapon/w_016.c [hunk 3]
```diff
@@ -133,7 +133,94 @@ s32 func_ptr_80170004(Entity* self) {
...
+            self->animSet += 2;
+        }
+        self->animCurFrame = 0xF;
+        self->facingLeft = PLAYER.facingLeft;
+        self->zPriority = PLAYER.zPriority - 2;
+        self->flags = FLAG_POS_CAMERA_LOCKED | FLAG_KEEP_ALIVE_OFFCAMERA;
+        self->drawFlags = DRAW_COLORS;
+
+        self->posY.i.hi = PLAYER.posY.i.hi + PLAYER.hitboxOffY;
+        if (PLAYER.step != 2) {
```
Comment (sozud) [inline src/weapon/w_016.c:162]: Player_Crouch

**Review by sozud — APPROVED** [review]
Nice work

**Pull request merged**